content-services/content-sources-backend
# Fixes 602: Send create notification

## Summary

Initial test of the notifications service integration. 

This current PR only implements the "BulkCreate" API.

A follow up PR will implement the rest of the notifications required after this pattern is proven.

## Testing steps

To test locally:  

- Navigate to "/pkg/notifications/client.go"

- comment out lines 19 - 25, and the closing brace on line 67

- Replace the "kafkaServers" on line 32 with the following:
<img width="1025" alt="Screen Shot 2023-05-11 at 10 24 36 AM" src="https://github.com/content-services/content-sources-backend/assets/38083295/3f12f7d3-a0cf-49a5-94ff-3a57bfd3020a">

- Install/run [KafkaCat](https://docs.confluent.io/platform/current/clients/kafkacat-usage.html#consumer-mode) so that you can record the message when it is sent.
- Call bulkCreate (through the API or via UI)
- Notice that kafkaCat should output a json blob. 
- (Optional) For schema verification, you can copy the json output from KafkaCat and paste it [here](https://internal.console.stage.redhat.com/api/notifications/internal/#/utils/console-cloud-event-validator), to very it is a valid schema.
